dataset/test/dataset_arithmetic_test.cpp

```diff
@@ -953,13 +769,19 @@ TEST(DatasetSetData, labels) {
 }
 
 TEST(DatasetInPlaceStrongExceptionGuarantee, events) {
-  auto good = make_events_variable_with_variance<double>();
-  set_events_values<double>(good, {{1, 2, 3}, {4}});
-  set_events_variances<double>(good, {{5, 6, 7}, {8}});
-  auto bad = make_events_variable_with_variance<double>();
-  set_events_values<double>(bad, {{0.1, 0.2, 0.3}, {0.4}});
-  set_events_variances<double>(bad, {{0.5, 0.6}, {0.8}});
+  Variable indicesGood = makeVariable<std::pair<scipp::index, scipp::index>>(
+      Dims{Dim::X}, Shape{2}, Values{std::pair{0, 2}, std::pair{2, 3}});
+  Variable indicesBad = makeVariable<std::pair<scipp::index, scipp::index>>(
+      Dims{Dim::X}, Shape{2}, Values{std::pair{0, 2}, std::pair{2, 4}});
+  Variable table =
+      makeVariable<double>(Dims{Dim::Event}, Shape{4}, units::m,
+                           Values{1, 2, 3, 4}, Variances{5, 6, 7, 8});
+  Variable good = from_constituents(indicesGood, Dim::Event, table);
+  Variable bad = from_constituents(indicesBad, Dim::Event, table);
```

Comment:
Why allow even to get this far when the bad indices are out of range?

Reply:
The indices are *not* out of range (`4` is "end", not "last"). They are bad only in the sense that they do not match the others.